client/resource_manager/client/model.go

```diff
@@ -83,20 +84,44 @@ func (kc *KVCalculator) BeforeKVRequest(consumption *rmpb.Consumption, req Reque
 	}
 }
 
+func (kc *KVCalculator) calculateWriteCost(consumption *rmpb.Consumption, req RequestInfo) {
+	writeBytes := float64(req.WriteBytes())
+	consumption.WriteBytes += writeBytes
+	consumption.WRU += float64(kc.WriteBaseCost) + float64(kc.WriteBytesCost)*writeBytes
+}
+
 // AfterKVRequest ...
 func (kc *KVCalculator) AfterKVRequest(consumption *rmpb.Consumption, req RequestInfo, res ResponseInfo) {
-	// For now, we can only collect the KV CPU cost for a read request.
 	if !req.IsWrite() {
-		kvCPUMs := float64(res.KVCPUMs())
-		consumption.TotalCpuTimeMs += kvCPUMs
-		consumption.RRU += float64(kc.CPUMsCost) * kvCPUMs
+		// For now, we can only collect the KV CPU cost for a read request.
+		kc.calculateCPUCost(consumption, res)
+	} else if !res.Succeed() {
+		// If the write request is not successfully returned, we need to pay back the WRU cost.
+		kc.payBackWriteCost(consumption, req)
 	}
 	// A write request may also read data, which should be counted into the RRU cost.
+	// This part should be counted even if the request does not succeed.
+	kc.calculateReadCost(consumption, res)
```

Comment:
Just curious, can this logic be leveraged by some one to intentionally issue volume of failing write requests, while no cost at all?

Reply:
Here only pay back the `write bytes` part.